great_expectations/validation_operators/types/validation_operator_result.py

```diff
@@ -0,0 +1,258 @@
+from typing import Dict, List, Union
+
+from great_expectations.core import ExpectationSuiteValidationResult
+from great_expectations.core.id_dict import BatchKwargs
+from great_expectations.data_context.types.resource_identifiers import (
+    ValidationResultIdentifier,
+)
+from great_expectations.types import DictDot
+
+
+class ValidationOperatorResult(DictDot):
+    """
+    The run_results property forms the backbone of this type and defines the basic contract for what a validation
+    operator's run method returns. It is a dictionary where the top-level keys are the ValidationResultIdentifiers of
+    the validation results generated in the run. Each value is a dictionary having at minimum, validation_result and
+    actions_results keys; this dictionary can contain other keys that are relevant for a specific validation operator
+    implementation. For example, the dictionary from a WarningAndFailureExpectationSuitesValidationOperator
+    would have an extra key named "expectation_suite_severity_level" to indicate if the suite is at either a
+    "warning" or "failure" level.
+
+    e.g.
+    {
```

Comment:
This is basic contract for what this type expects to get from a `ValidationOperator` run. Some notes/?'s:
- top-level keys are just the `ValidationResultIdentifiers` from the run
- corresponding value is a dict with at least a `validation_result` key with the actual `ExpectationSuiteValidationResult` and an `actions_results` key (this is dict of all action outputs from validation)
- all other properties are derived from previous, lazily
- new validation operators may add additional keys here as needed, in which case their values would only be accessible from the `run_results` key

- right now, the only two validation operators are `action_list` based, hence the inclusion of the `actions_results` key in the basic contract. I can imagine this not being true in the future though. Should we keep this in the basic contract? If not, it would just mean the `actions_results_by_validation_result_identifier` property goes away and is only accessible from the `run_results` property

Reply:
Basic contract too. There maybe folks who are using actions_results_by_validation_result_identifier currently.